.gitlab/build.sh

```
@@ -163,30 +162,46 @@ else
+    # DOCA + RDMA build dependencies.
+    #  - Bases without DOCA (cuda-dl-base, nvidia/cuda, ubuntu22.04): add the DOCA
+    #    3.3.0 host repo, install the SDK + headers, then reinstall the RDMA packages
+    #    to repair cuda-dl-base's broken libibverbs-dev.
+    #  - Bases that already ship DOCA (nvcr.io/nvidia/pytorch bundles >=3.4): use that
+    #    stack as-is. Adding the older 3.3 repo would only downgrade/mismatch it, so
+    #    skip the whole repo add + SDK install + RDMA reinstall.
+    if dpkg -s doca-sdk-gpunetio >/dev/null 2>&1; then
+        echo "DOCA $(dpkg-query -W -f='${Version}' doca-sdk-gpunetio) provided by base image; skipping DOCA 3.3 repo, SDK install, and RDMA reinstall"
+    else
+        ARCH_SUFFIX=$(if [ "${ARCH}" = "aarch64" ]; then echo "arm64"; else echo "amd64"; fi)
+        MELLANOX_OS="$(. /etc/lsb-release; echo ${DISTRIB_ID}${DISTRIB_RELEASE} | tr A-Z a-z | tr -d .)"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)